### PR TITLE
Add rest interface [Do Not Merge]

### PR DESCRIPTION
**Pull request by kmansel**
Please don't merge this yet, i'm working on the piece to split out to each port.

### CODE REVIEW

**Comment by jfelten** [comment]
@alejandroEsc fails CI

$ protoc -I api/ api/api.proto --go_out=plugins=grpc:api
google/api/annotations.proto: File not found.
api.proto: Import "google/api/annotations.proto" was not found or had errors.

**Comment by kmansel** [comment]
Ok, i think we're ready to go here, @leahnp @alejandroEsc @venezia would any of you like to take a quick look?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -41,7 +41,9 @@
   "script":
   - "go get -u google.golang.org/grpc"
   - "go get -u github.com/golang/protobuf/protoc-gen-go"
-  - "protoc -I api/ api/api.proto --go_out=plugins=grpc:api"
+  - "go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
+  - "protoc -I api/ api/api.proto -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --go_out=plugins=grpc:api"
```
Comment (alejandroEsc) [inline .gitlab-ci.yml]: havent checked the image being used, but I am guessing this should work?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,55 @@
-# sample-grpc-apiserver
-gRPC API POC
+# Setting Up Your Environment
+Keep in mind that gRPC requires golang v1.6 or greater, to install run
+
```
Comment (alejandroEsc) [inline README.md]: nit, inconsistent spacing here, not going to be somethign i wont merge for, just saying after a line either have a space then the code, or no space. I prefer the latter, but again doesnt matter which you choose.

File: api/api.proto [hunk 3]
```diff
@@ -16,12 +16,22 @@ syntax = "proto3";
 
 package trident;
 
+import "google/api/annotations.proto";
```
Comment (alejandroEsc) [inline api/api.proto]: can you add some swagger annotation also?

File: server/main.go [hunk 4]
```diff
@@ -1,22 +1,42 @@
+/*
```
Comment (alejandroEsc) [inline server/main.go]: good call on adding this.

File: server/main.go [hunk 5]
```diff
@@ -1,22 +1,42 @@
...
 	"os/signal"
 	"syscall"
 	"time"
-
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 	 pb "../api"
+	"net/http"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"net"
```
Comment (alejandroEsc) [inline server/main.go]: not your fault here but when i add the tool ./bin/clean.sh make the imports should be cleared up.

File: server/main.go [hunk 6]
```diff
@@ -1,22 +1,42 @@
...
 	"google.golang.org/grpc/reflection"
 )
 
 var (
-  serverPort = flag.String("server_port", ":5300", "Server port")
+  	serverAddrGRPC = flag.String("server_addr_grpc", "127.0.0.1:5300", "The server address in the format of host:port")
```
Comment (alejandroEsc) [inline server/main.go]: I am going to remove flags later on, hope this is ok? Factor 12 apps do not use flags.

File: .gitlab-ci.yml (hunk 1 above)
Comment (kmansel) [reply .gitlab-ci.yml]: yes this works, i had to put it in so that it would create the correct gateway protobuf file

File: README.md (hunk 2 above)
Comment (kmansel) [reply README.md]: Ah...no worries, i'll take a look at the readme and make sure it's all good to go.

File: api/api.proto (hunk 3 above)
Comment (kmansel) [reply api/api.proto]: do we want to add that at this time?  I think we'd have to create another mux to handle the swagger stuff right?  Would that be a good ticket as a feature request?

File: server/main.go (hunk 4 above)
Comment (kmansel) [reply server/main.go]: 👍

File: server/main.go (hunk 5 above)
Comment (kmansel) [reply server/main.go]: Noted.  Will gofmt the files before sending them up again.  😄

File: server/main.go (hunk 6 above)
Comment (kmansel) [reply server/main.go]: It's a default, so however we want to put that in, a const if we want it hard coded or just a overridable variable...that would be a design question...most likely following the 12 factor route of throwing it into env vars and running it that way with a fallback default is my guess.

**Comment by kmansel** [comment]
@alejandroEsc I believe i've addressed your list here.  👍

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -41,7 +41,9 @@
   "script":
```
Comment (leahnp) [inline .gitlab-ci.yml]: Can we make these non strings? I was having a few weird glitches in CI due to them. Check out what I've done with the pipeline in my PR: https://github.com/leahnp/sample-grpc-apiserver/blob/2f3677a682e2b50645ffa2bcd9a2e67aff681fc9/.gitlab-ci.yml let me know if you think it makes sense/is worth it to clean up.

File: api/api.proto (hunk 3 above)
Comment (leahnp) [reply api/api.proto]: I think this is out of scope for this ticket.

File: .gitlab-ci.yml (hunk 1 above)
Comment (leahnp) [reply .gitlab-ci.yml]: Can confirm it works - I am also using it for my job

**Comment by alejandroEsc** [comment]
lgtm, merging.

**Pull request merged**